lua/lib/lib.lua

```diff
@@ -120,12 +120,17 @@ end
 function M.refresh_tree()
   -- local stat = luv.fs_stat(M.Tree.cwd)
   -- if stat.mtime.sec ~= M.Tree.last_modified then
-    refresh_nodes(M.Tree)
+  refresh_nodes(M.Tree)
   -- end
+  local _, status = git.get_git_root(M.Tree.cwd)
+
   if config.get_icon_state().show_git_icon or vim.g.lua_tree_git_hl then
-    git.reload_roots()
-    refresh_git(M.Tree)
+    if status ~= git.not_git then
+      git.reload_roots()
+      refresh_git(M.Tree)
+    end
   end
+
```

Comment:
did you test the case where:
your root is not a git repository, but some folders inside are git repositories :) ?

Reply:
So I've simplified quite a bit by just saying if the root of the tree isn't a git directory then don't recurse through the branches

Reply:
I'm not actually sure how we would keep the aucmd fast and still recurse through all of the branches? I'm basically assuming that the majority case use of vim is in a project directory that's initialised for git or for making configuration changes in potentially vast trees. The truth is that that assumption won't cover some use cases and therefore people will be losing out functionality for speed.